test/test_distributed_transpose.jl

```diff
@@ -0,0 +1,52 @@
+using MPI
+
+MPI.Init()
+
+include("dependencies_for_runtests.jl")
+
+using Oceananigans.DistributedComputations: 
+                ParallelFields, 
+                transpose_z_to_y!, 
+                transpose_y_to_z!,
+                transpose_y_to_x!,
+                transpose_x_to_y!
+
+function test_transpose(grid_points, ranks, topo)
+    arch = Distributed(CPU(), ranks=ranks, topology=topo)
+    grid = RectilinearGrid(arch, topology=topo, size=grid_points, extent=(2π, 2π, 2π))
+
+    ϕ = CenterField(grid)
+    Φ = ParallelFields(ϕ)
```

Comment:
What's this?

Reply:
It's a "container" to perform transposes without needing memory allocation. It includes a `xfield` (a field on a x-free configuration) a `yfield` (a field on a y-free configuration) and a `zfield` (a field on a z-grid configuration) plus the correctly sized buffers and partial communicators needed for the transpose.

the idea is that if you call
```julia
transpose_y_to_z!(Φ::ParallelFields)
```

you will transpose the data from `Φ.yfield` to `Φ.zfield`.

Reply:
we can call it `TransposeHelper` or `TransposeFields`

Reply:
What does the `s` refer to? Eg why is this plural.

Reply:
Should it be `TransposableField`? (no "s")